pkg/cmd/grafana-cli/commands/list_upgradeable_command.go

```diff
@@ -0,0 +1,47 @@
+package commands
+
+import (
+	"errors"
+
+	"github.com/grafana/grafana/pkg/cmd/grafana-cli/logger"
+	"github.com/grafana/grafana/pkg/cmd/grafana-cli/services"
+	"github.com/grafana/grafana/pkg/cmd/grafana-cli/utils"
+	"github.com/hashicorp/go-version"
+)
+
+func (cmd Command) listUpgradeableCommand(c utils.CommandLine) error {
+	pluginsDir := c.PluginDirectory()
+
+	localPlugins := services.GetLocalPlugins(pluginsDir)
+
+	remotePlugins, err := cmd.Client.ListAllPlugins(c.String("repo"))
+	if err != nil {
+		return err
+	}
+
+	anythingToUpgrade := false
+
+	for _, localPlugin := range localPlugins {
+		for _, p := range remotePlugins.Plugins {
+			remotePlugin := p
+			if localPlugin.ID != remotePlugin.ID {
+				continue
+			}
+			if shouldUpgrade(localPlugin.Info.Version, &remotePlugin) {
+				latest := latestSupportedVersion(&remotePlugin)
+				latestVersion, err := version.NewVersion(latest.Version)
+				if err != nil {
+					return err
+				}
+				anythingToUpgrade = true
+				logger.Infof("%v is upgradeable (%v -> %v)\n", localPlugin.ID, localPlugin.Info.Version, latestVersion)
+			}
+		}
+	}
+
+	if anythingToUpgrade {
+		return nil
+	} else {
+		return errors.New("no plugins are upgradeable")
```

Comment:
I don't think an error is appropriate here. A log message "no plugins are upgradeable" would be good though

Reply:
Nothing short of an error makes us exit 1 and that is needed to be easily able to detect the outcome of the command.

Any ideas how we can exit 1 and not return an error?

Reply:
Is this the only command you're using like that? Because AFAIK none of the other commands operate on that style

Reply:
This is the reason I want to implement this command since it will behave this way. "Parsing" exit status is a much easier way to determine if we need/can upgrade the plugins than parsing an unstable and unstructured output of multiple commands (loop over `ls` and to `remote-versions` on every one and check if that is newer). So much easier to add another command which actually does that.